addons/stock_dropshipping/stock_dropshipping.py

```diff
@@ -0,0 +1,36 @@
+# -*- coding: utf-8 -*-
+
+from openerp import models, api, _
+from openerp.exceptions import Warning
+
+class stock_warehouse(models.Model):
+    _inherit = 'stock.warehouse'
+
+    @api.model
+    def _get_drop_shipping_route(self):
+        try:
+            ds_route_id = self.env.ref("stock_dropshipping.route_drop_shipping")
+        except:
+            ds_route_id = self.env['stock.location.route'].search([('name', 'like', _('Drop Shipping'))])
+            ds_route_id = ds_route_id and ds_route_id[0] or False
+        if not ds_route_id:
+            raise Warning(_('Can\'t find any generic Drop Shipping route.'))
+        return ds_route_id.id
+
+class sale_order_line(models.Model):
+    _inherit = 'sale.order.line'
+
```

Comment:
@jco-odoo I don't think I get your above comment completely, because we moved some stuff from "product_id_change_with_wh(...)" to "_check_routing(...)" so that we can override it in stock_dropshipping module and hide the "Not Enough Stock !" warning, which basically invoked on changing the product in sale order line.
But what I don't get completely is the thing you want to do with "route_id" in sale order line. I don't think we could do something for that warning based on "route_id" field of sale order line. So, would you please shed light on this.